src/main/java/games/strategy/triplea/player/ITripleAPlayer.java

```diff
@@ -166,6 +166,17 @@ Unit whatShouldBomberBomb(Territory territory, final Collection<Unit> potentialT
   Territory whereShouldRocketsAttack(Collection<Territory> candidates, Territory from);
 
   /**
+   * Choose which territory to produce units from. Used when multiple factories are available for sea zones.
+   * 
+   * @param candidates
+   *        - a collection of Territories, the possible territories to produce units
+   * @param unitTerritory
+   *        - where units are being placed
+   * @return the territory selected as the producer
```

Comment:
Might want to document that `null` can be returned if no territory is selected.

Alternatively, you could return `Optional<Territory>`.